appcelerator/titanium_mobile
# fix(ios): xcode project change detection causing unnecessary rebuilds

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28506

**Summary:**
This fixes two issues where our build was unnecessarily doing full Xcode re-builds...

- Always integrate frameworks into into the Xcode project in the same order. Otherwise the Xcode project file is different from the last run. Switched from iterating a map to a sorted array to ensure order. Also fixed an issue with loading previous state data which caused the framework hook to always scan for available frameworks inside the project.
- Trim whitespaces when comparing generated `.entitlements` with existing ones from Xcode extensions before overwriting. Otherwise they may differ just by a trailing newline for example and always cause a full re-build.

**Test:**
1. Build [kitchensink-v2](https://github.com/appcelerator/kitchensink-v2) for iOS.
2. Clear the log.
3. Build the project for iOS again without making any changes.
4. Verify that you do **NOT** see the following in the log. _(This is the fix.)_
```
Forcing rebuild: Xcode project has changed since last build
```
